File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Power/ThieveryPower.ts
import { Mathf } from "../../../../Core/Module/Math/Mathf";
import DungeonManager from "../DungeonManager";
import { OnDeathMessage } from "../Events/OnDeathMessage";
import FileHelper from "../FileHelper";
import { LocalizationProperty } from "../Gen/DB/Localization";
import { AbstractPower, PowerType } from "./AbstractPower";

export class ThieveryPower extends AbstractPower{
    public static Id = "Thievery"; 
    Icon: string = FileHelper.FormatPath("Powers/thievery.png");
    Id: string = "Thievery";
    Name: string = LocalizationProperty.Read("能力-偷窃");
    get Type(): PowerType {
        return PowerType.Buff;
    }

    GetDescription(...args: any[]): string {
        return LocalizationProperty.Read("能力-偷窃描述");
    }

    protected OnDeath(msg: OnDeathMessage) {
        super.OnDeath(msg);
        DungeonManager.Inst.CurrentRoom.AddReward(Mathf.Clamp(this.Amount,0,Number.MAX_SAFE_INTEGER));
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Power/Special/ThieveryPower.cs
using ModTheSpire.Dungeon;
using ModTheSpire.Event.Ids;
using TCG.Config;
using UnityEngine;

namespace ModTheSpire.Power.Buff
{
    public class ThieveryPower : AbstractPower
    {
        public override string Id { get; } = nameof(ThieveryPower);
        public override string Name { get; } = LocalizationProperty.Read("能力-偷窃");
        public override string Icon { get; } = "Powers/thievery.png";
        public override PowerType Type { get; } = PowerType.Special;
        public override string GetDescription(object[] args)
        {
            return LocalizationProperty.Read("能力-偷窃描述");
        }

        protected override void OnDeath(OnDeathMessage msg)
        {
            base.OnDeath(msg);
            DungeonManager.Inst.CurrentRoom.AddReward(new object[] { Mathf.Clamp(this.Amount, 0, int.MaxValue) });
        }
    }
}